dolphinscheduler-e2e/dolphinscheduler-e2e-case/src/test/java/org/apache/dolphinscheduler/e2e/pages/project/workflow/task/JavaTaskForm.java

```diff
@@ -17,27 +17,111 @@
 
 package org.apache.dolphinscheduler.e2e.pages.project.workflow.task;
 
-import org.apache.dolphinscheduler.e2e.pages.common.CodeEditor;
 import org.apache.dolphinscheduler.e2e.pages.project.workflow.WorkflowForm;
 
+import java.time.Duration;
+import java.util.List;
+
+import lombok.Getter;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.Keys;
 import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
 
+@Getter
 public class JavaTaskForm extends TaskNodeForm {
 
-    private CodeEditor codeEditor;
-
     private WebDriver driver;
 
+    @FindBy(xpath = "//label[span[text()='Run Type']]/following-sibling::div//div[contains(@class, 'n-base-selection')]")
+    private WebElement selectRunType;
+
+    @FindBy(xpath = "//label[span[text()='Main Package']]/following-sibling::div//div[contains(@class, 'n-base-selection-label')]")
+    private WebElement selectMainPackage;
+
+    @FindBy(xpath = "//div[@class='n-form-item-blank']//div[contains(@class, 'n-base-selection--multiple')]//div[contains(@class, 'n-base-selection-tags') and @tabindex='0']")
+    private WebElement selectResource;
+
     public JavaTaskForm(WorkflowForm parent) {
         super(parent);
 
-        this.codeEditor = new CodeEditor(parent.driver());
-
         this.driver = parent.driver();
+
+        PageFactory.initElements(driver, this);
     }
 
-    public JavaTaskForm script(String script) {
-        codeEditor.content(script);
+    public TaskNodeForm selectResource(String resourceName) {
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(60));
```

Comment:
Please using `WebDriverWaitFactory.createWebDriverWait(page.driver())` instead.

Reply:
The code has been modified